disassemblers/ddisasm/ddisasmBBorRef.py

```diff
@@ -0,0 +1,305 @@
+                        kind = 0 # TODO
+
+                    ref = refInf.ref.add()
+                    update_ref(m, ref, from_addr, to_addr, kind)
+
+                    logging.debug(
+                        "%d: From 0x%x -> 0x%x"
+                        % (kind, ref.ref_va, ref.target_va)
+                    )
+
+    logging.info("Collect Refs done! ready to write output...")
+    f = open(output, "wb")
+    f.write(refInf.SerializeToString())
+    f.close()
+
+
+class FileType(Enum):
+    IR = "ir"
+    ELF = "elf"
+    PE = "pe"
+    OTHERS = "others"
+
+
+def get_magic_file_type(file_path: Path) -> FileType:
+    """
+    Get the FileType of the given file
+    """
+    with open(file_path, "rb") as magic_fp:
+        magic = magic_fp.read(5)
+        if magic[:5] == b"GTIRB":
+            return FileType.IR
+        elif magic[:4] == b"\x7fELF":
+            return FileType.ELF
+        else:
+            dos_header = magic_fp.read(64)
+            if len(dos_header) < 64 or dos_header[:2] != b'MZ':
+                return FileType.OTHERS
+            pe_offset = int.from_bytes(dos_header[60:64], byteorder='little')
+            magic_fp.seek(pe_offset)
+            if magic_fp.read(4) == b'PE\x00\x00':
+                return FileType.PE
+            else:
+                return FileType.OTHERS
+
+
+class ExtractMode(Enum):
+    BB = "bb"
+    REF = "ref"
+
+
+if __name__ == "__main__":
+    parser = optparse.OptionParser()
+    parser.add_option(
+        "-m", "--mode",
+        metavar="string", 
+        help="Select extract-mode: `bb` or `ref`",
+        default=ExtractMode.BB.value,
+    )
+    parser.add_option(
+        "-i", "--input",
+        action="store",
+        type="string",
+        help="input binary or gtirb-IR file",
+        default=None
+    )
+    parser.add_option(
+        "-o", "--output",
+        action="store",
+        type="string",
+        help="output of the protobuf file",
+        default=None
+    )
+    (options, args) = parser.parse_args()
+
+    if options.input is None:
+        print("Please provide an input binary or gtirb-IR file")
+        exit(-1)
+
+    if options.output is not None:
+        pb_output = Path(options.output)
+    else:
+        pb_output = Path("/tmp") / f"{Path(options.input).name}.ddisasm.pb"
+
+    file_type = get_magic_file_type(options.input)
+
+    if file_type == FileType.ELF or file_type == FileType.PE:
+        logging.info("Run ddisasm on %s", options.input)
+        start = time.monotonic()
+        ir_path = Path(pb_output).with_suffix(".gtirb")
+        disassemble(Path(options.input), ir_path)
+        logging.info("Done with ddisasm.")
+        end = time.monotonic()
+        logging.info(f"time-for-ddisasm: {options.input}: {end - start:.2f}")
+    elif file_type == FileType.IR:
+        ir_path = options.input
+    else:
+        logging.warning(f"Unknown file type: %s", options.input)
+        exit(-1)
```

Comment:
why -1? I would usually expect `exit(1)` for something like this. Same for other -1s

Reply:
Fixed.